uniffi_macros/src/export/item.rs

```diff
@@ -82,9 +83,14 @@ impl ExportItem {
 
                 let attrs = ExportedImplFnAttributes::new(&impl_fn.attrs)?;
                 let item = if attrs.constructor {
-                    ImplItem::Constructor(ConstructorSignature::new(impl_fn.sig)?)
+                    let sig = FnSignature::new_constructor(self_ident.clone(), impl_fn.sig)?;
+                    if sig.is_async {
+                        return sig.syn_err("Async constructors are not supported");
+                    }
+                    ImplItem::Constructor(sig)
```

Comment:
Why is the `is_async` check not part of `new_constructor`? Also, having a `syn_err` method seems kind of weird to me.